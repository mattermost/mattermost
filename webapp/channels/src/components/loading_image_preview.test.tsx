// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {screen, render} from '@testing-library/react';

import LoadingImagePreview from 'components/loading_image_preview';

describe('components/LoadingImagePreview', () => {
    test('should render loading text with progress', () => {
        const loading = 'Loading';
        let progress = 50;

        const {rerender} = render(
            <LoadingImagePreview
                loading={loading}
                progress={progress}
            />,
        );

        expect(screen.getByText('Loading 50%')).toBeInTheDocument();

        progress = 90;
        rerender(
            <LoadingImagePreview
                loading={loading}
                progress={progress}
            />,
        );

        expect(screen.getByText('Loading 90%')).toBeInTheDocument();
    });
});
