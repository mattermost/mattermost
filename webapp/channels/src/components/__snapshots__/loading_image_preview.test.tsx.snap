// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoadingImagePreview should match snapshot 1`] = `
<div
  className="view-image__loading"
>
  <Memo(LoadingSpinner) />
  <span
    className="loader-percent"
  >
    Loading
     50%
  </span>
</div>
`;

exports[`components/LoadingImagePreview should match snapshot 2`] = `
<div
  className="view-image__loading"
>
  <Memo(LoadingSpinner) />
  <span
    className="loader-percent"
  >
    Loading
     90%
  </span>
</div>
`;
